### PR TITLE
Add Material 3 `FloatingActionButton` and `FloatingActionButton` variants examples

### PR DESCRIPTION
**Pull request by TahaTesser**
part of https://github.com/flutter/flutter/issues/101048

| M3 FAB | M3 FAB Variants |
| --------------- | --------------- |
|  |  |

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/floating_action_button/floating_action_button.2.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'FloatingActionButton Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
```
Comment (guidezpl) [inline examples/api/lib/material/floating_action_button/floating_action_button.2.dart:20]: I'll see if I can get the seed color for the spec for we match

File: examples/api/lib/material/floating_action_button/floating_action_button.2.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(title),
+      ),
+      body: const Center(child: Text('Press the button below!')),
+      // An example of the floating action button.
+      //
+      // https://m3.material.io/components/floating-action-button/specs#71504201-7bd1-423d-8bb7-07e0291743e5
```
Comment (guidezpl) [inline examples/api/lib/material/floating_action_button/floating_action_button.2.dart:40]:
```suggestion
      // https://m3.material.io/components/floating-action-button/specs
```

File: examples/api/lib/material/floating_action_button/floating_action_button.2.dart (hunk 1 above)
Comment (TahaTesser) [reply examples/api/lib/material/floating_action_button/floating_action_button.2.dart:20]: That would be cool, I just matched card example seed color.

Comment (guidezpl) [reply examples/api/lib/material/floating_action_button/floating_action_button.2.dart:20]: good guess ;)

**Comment by TahaTesser** [comment]
cc: @gspencergoog

File: examples/api/lib/material/floating_action_button/floating_action_button.0.dart [hunk 3]
```diff
@@ -11,25 +11,27 @@ void main() => runApp(const MyApp());
...
       title: _title,
-      home: MyStatelessWidget(),
+      home: FABExample(title: _title),
     );
   }
 }
 
-class MyStatelessWidget extends StatelessWidget {
-  const MyStatelessWidget({Key? key}) : super(key: key);
+class FABExample extends StatelessWidget {
```
Comment (gspencergoog) [inline examples/api/lib/material/floating_action_button/floating_action_button.0.dart:25]: Probably should be named `FabExample` to stay with the naming guidelines.

File: examples/api/lib/material/floating_action_button/floating_action_button.3.dart [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
+      home: const FABExample(title: _title),
+    );
+  }
+}
+
+class FABExample extends StatelessWidget {
```
Comment (gspencergoog) [inline examples/api/lib/material/floating_action_button/floating_action_button.3.dart:26]: Probably should be named `FabExample`.

File: examples/api/lib/material/floating_action_button/floating_action_button.1.dart [hunk 5]
```diff
@@ -11,25 +11,27 @@ void main() => runApp(const MyApp());
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
 
-  static const String _title = 'Flutter Code Sample';
+  static const String _title = 'FloatingActionButton Sample';
 
   @override
   Widget build(BuildContext context) {
     return const MaterialApp(
       title: _title,
```
Comment (gspencergoog) [inline examples/api/lib/material/floating_action_button/floating_action_button.1.dart:19]: We don't actually need an app title (it only sets the app title for the task on Android, I think), so you could eliminate the `_title` variable, and the argument to `FabExample`, and just hard code the string in the AppBar.

(and in the other examples too).

**Comment by TahaTesser** [comment]
Confusing this PR with similar M3 doc PRs, added tree label by mistake
This still needs approval @gspencergoog

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**